File: client/src/api/env-config.ts
/**
 * API client for environment configuration management
 */

import { apiClient } from '../lib/api';

export interface EnvVarResponse {
  key: string;
  value: string;
  is_sensitive: boolean;
  category: string;
  description_key: string;
  input_type: string; // text | number | boolean | secret
  default: string | null;
  file: string; // backend | client
}

export interface EnvConfigResponse {
  backend: EnvVarResponse[];
  client: EnvVarResponse[];
  categories: string[];
}

export interface EnvVarUpdate {
  key: string;
  value: string;
}

export interface EnvConfigUpdateRequest {
  file: 'backend' | 'client';
  updates: EnvVarUpdate[];
}

export interface EnvConfigUpdateResponse {
  changed: string[];
  count: number;
}

export interface EnvVarRevealResponse {
  key: string;
  value: string;
}

export async function getEnvConfig(): Promise<EnvConfigResponse> {
  const response = await apiClient.get<EnvConfigResponse>('/api/admin/env-config');
  return response.data;
}

export async function updateEnvConfig(request: EnvConfigUpdateRequest): Promise<EnvConfigUpdateResponse> {
  const response = await apiClient.put<EnvConfigUpdateResponse>('/api/admin/env-config', request);
  return response.data;
}

export async function revealEnvVar(key: string): Promise<string> {
  const response = await apiClient.get<EnvVarRevealResponse>(`/api/admin/env-config/reveal/${encodeURIComponent(key)}`);
  return response.data.value;
}

File: backend/app/schemas/env_config.py
"""Pydantic schemas for environment configuration management."""

import re
from typing import Optional
from pydantic import BaseModel, field_validator


class EnvVarResponse(BaseModel):
    """Single environment variable (sensitive values masked)."""
    key: str
    value: str
    is_sensitive: bool
    category: str
    description_key: str
    input_type: str  # text | number | boolean | secret
    default: Optional[str] = None
    file: str  # backend | client


class EnvConfigReadResponse(BaseModel):
    """Full env config read response."""
    backend: list[EnvVarResponse]
    client: list[EnvVarResponse]
    categories: list[str]


class EnvVarUpdate(BaseModel):
    """Single variable update."""
    key: str
    value: str

    @field_validator("key")
    @classmethod
    def validate_key(cls, v: str) -> str:
        if not re.match(r"^[A-Z][A-Z0-9_]*$", v):
            raise ValueError("Key must be uppercase letters, digits, and underscores")
        return v


class EnvConfigUpdateRequest(BaseModel):
    """Update request for one or more variables in a specific file."""
    file: str  # "backend" | "client"
    updates: list[EnvVarUpdate]

    @field_validator("file")
    @classmethod
    def validate_file(cls, v: str) -> str:
        if v not in ("backend", "client"):
            raise ValueError("file must be 'backend' or 'client'")
        return v


class EnvVarRevealResponse(BaseModel):
    """Response for revealing a single sensitive value."""
    key: str
    value: str
